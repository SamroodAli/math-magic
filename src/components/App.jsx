import {
  BrowserRouter as Router, Link,
} from 'react-router-dom';

import React, { useState } from 'react';
import { Grid } from 'semantic-ui-react';
import NavMenu from './NavMenu';
import Routes from './Routes';
import './style/App.css';

const App = () => {
  const [activeItem, setActiveItem] = useState('Home');
  const handleItemClick = (e, { name }) => setActiveItem(name);

  return (
    <Router>
      <div className="ui center aligned container">
        <header>
          <Link to="/">
            <h1 className="text-center" id="brand">Math Magic</h1>
          </Link>
        </header>
        <Grid>
          <Grid.Column width={4}>
            <NavMenu activeItem={activeItem} handleItemClick={handleItemClick} />
          </Grid.Column>
          <Grid.Column stretched width={12}>
            <Routes />
          </Grid.Column>
        </Grid>
      </div>
    </Router>
  );
};

export default App;
