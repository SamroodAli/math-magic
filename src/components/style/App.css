.container {
  height: 100vh;
}

#brand {
  font-size: 3rem;
  margin-bottom: 1rem;
}
