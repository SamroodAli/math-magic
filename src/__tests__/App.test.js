import renderer from 'react-test-renderer';
import App from '../components/App';

test('Button snapshot testing', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('App renders display component', () => {});
