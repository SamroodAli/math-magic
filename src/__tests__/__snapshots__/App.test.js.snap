// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot testing 1`] = `
Array [
  <p>
    Total: 0
  </p>,
  <p>
    undefined: 
  </p>,
  <p>
    Result:: 0
  </p>,
  <div>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      รท
    </button>
  </div>,
  <div>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>,
  <div>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>,
  <div>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>,
  <div>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      data-testid="button"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>,
]
`;
